app/org/sagebionetworks/bridge/models/healthdata/HealthDataUserConsent.java

```diff
@@ -0,0 +1,85 @@
+package org.sagebionetworks.bridge.models.healthdata;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import org.joda.time.DateTime;
+
+import org.sagebionetworks.bridge.json.DateTimeToStringSerializer;
+import org.sagebionetworks.bridge.json.JodaDateTimeDeserializer;
+
+/**
+ * User consent metadata stored with the health data record. We store this in the Health Data Record table, so we can
+ * avoid doing a bunch of extra calls to other tables during export.
+ */
+@JsonDeserialize(builder = HealthDataUserConsent.Builder.class)
+public class HealthDataUserConsent {
```

Comment:
I understand the benefits for performance.  I'm not clear about the cost of the maintaining a copy of user consent. For example, if the user turned off sharing, shouldn't this copy be searched and modified accordingly?  Another example is about maintaining the code.  Alx just changed sharing from a boolean to an enum.  This class is an additional place that needs to be changed accordingly.